api/database/factories/PoolFactory.php

```diff
@@ -72,57 +71,93 @@ public function definition()
         ];
     }
 
-    public function configure()
+    public function withPoolSkills($essentialCount, $nonEssentialCount)
     {
-        return $this->afterCreating(function (Pool $pool) {
+        return $this->afterCreating(function (Pool $pool) use ($essentialCount, $nonEssentialCount) {
+            $this->$essentialCount = $essentialCount;
+            $this->$nonEssentialCount = $nonEssentialCount;
+
             $skills = Skill::inRandomOrder()->limit(10)->get();
+            //slice first set of skills as essential skills
+            $essentialSkills = $skills->slice(0, $essentialCount);
+            //slice next set of  skills as non essential skills
+            $nonEssentialSkills = $skills->slice($essentialCount, $nonEssentialCount);
+            $this->createPoolSkills($pool, $essentialSkills, PoolSkillType::ESSENTIAL->name);
+            $this->createPoolSkills($pool, $nonEssentialSkills, PoolSkillType::NONESSENTIAL->name);
+        });
+    }
 
-            foreach ($skills->slice(0, 5) as $skill) {
```

Comment:
I've removed this hard coded skills during pool creation as we need diff no. of skills for the tests.